test/myxql/protocol/values_test.exs

```diff
@@ -274,15 +274,119 @@ defmodule MyXQL.Protocol.ValueTest do
+      # end
+
+      # @tag geometry: true
+      # test "POINTZM", c do
+      #   assert_roundtrip(c, "my_point", %Geo.PointZM{coordinates: {1, 1, 1, 1}})
+      # end
+      #
+      @tag geometry: true
+      test "LINESTRING", c do
+        assert_roundtrip(c, "my_linestring", %Geo.LineString{
+          coordinates: [{30, 10}, {10, 30}, {40, 40}]
+        })
       end
 
+      # @tag geometry: true
+      # test "LINESTRINGZ", c do
+      #   assert_roundtrip(c, "my_linestring", %Geo.LineStringZ{
+      #     coordinates: [{30, 10, 3}, {10, 30, 90}, {40, 40, 40}]
+      #   })
+      # end
+
+      @tag geometry: true
       test "MULTIPOINT", c do
         assert_roundtrip(c, "my_multipoint", %Geo.MultiPoint{
           coordinates: [{1.0, 1.0}, {2.0, 2.0}]
         })
       end
+
+      # @tag geometry: true
+      # test "MULTIPOINTZ", c do
+      #   assert_roundtrip(c, "my_multipoint", %Geo.MultiPointZ{
+      #     coordinates: [{0, 0, 0}, {20, 20, 20}, {60, 60, 60}]
+      #   })
+      # end
+
+      @tag geometry: true
+      test "MULTILINESTRING", c do
+        assert_roundtrip(c, "my_multilinestring", %Geo.MultiLineString{
+          coordinates: [[{10, 10}, {20, 20}], [{15, 15}, {30, 15}]]
+        })
+      end
+
+      # @tag geometry: true
+      # test "MULTILINESTRINGZ", c do
+      #   assert_roundtrip(c, "my_multilinestring", %Geo.MultiLineStringZ{
+      #     coordinates: [[{10, 10, 10}, {20, 20, 20}], [{15, 15, 15}, {30, 15, 10}]]
+      #   })
+      # end
+
+      @tag geometry: true
+      test "POLYGON", c do
+        assert_roundtrip(c, "my_polygon", %Geo.Polygon{
+          coordinates: [[{30, 10}, {40, 40}, {20, 40}, {10, 20}, {30, 10}]]
+        })
+      end
+
+      # @tag geometry: true
+      # test "POLYGONZ", c do
+      #   assert_roundtrip(c, "my_polygon", %Geo.PolygonZ{
+      #     coordinates: [
+      #       [{35, 10}, {45, 45}, {15, 40}, {10, 20}, {35, 10}],
+      #       [{20, 30}, {35, 35}, {30, 20}, {20, 30}]
+      #     ]
+      #   })
+      # end
+
+      @tag geometry: true
+      test "MULTIPOLYGON", c do
+        assert_roundtrip(c, "my_multipolygon", %Geo.MultiPolygon{
+          coordinates: [
+            [[{40, 40}, {20, 45}, {45, 30}, {40, 40}]],
+            [
+              [{20, 35}, {10, 30}, {10, 10}, {30, 5}, {45, 20}, {20, 35}],
+              [{30, 20}, {20, 15}, {20, 25}, {30, 20}]
+            ]
+          ]
+        })
+      end
+
+      # @tag geometry: true
+      # test "MULTIPOLYGONZ", c do
+      #   assert_roundtrip(c, "my_multipolygon", %Geo.MultiPolygonZ{
+      #     coordinates: [
+      #       [[{40, 40, 40}, {20, 45, 60}, {45, 30, 15}, {40, 40, 40}]],
+      #       [
+      #         [{20, 35, 48}, {10, 30, 50}, {10, 10, 10}, {30, 5, 18}, {45, 20, 10}, {20, 35, 48}],
+      #         [{30, 20, 10}, {20, 15, 10}, {20, 25, 4}, {30, 20, 10}]
+      #       ]
+      #     ]
+      #   })
+      # end
+
+      if @protocol == :text do
```

Comment:
I tried replying over email and doesn't look like GitHub UI shows my response so trying again here:

> We’re missing `def encode_binary_value(%Geo.GeometryCollection{} = geo), do: encode_geometry(geo)`, after that we should be good to go.

The error you were getting about Jason is we attempt to encode lists and maps to the json column, but because structs are maps that clause got invoked.